lib/dry/cli.rb

```diff
@@ -135,23 +132,50 @@ def parse(command, arguments, names)
 
       result = Parser.call(command, arguments, prog_name)
 
-      if result.help?
-        out.puts Banner.call(command, prog_name)
-        exit(0)
-      end
+      return help(command, prog_name) if result.help?
 
-      if result.error?
-        err.puts(result.error)
-        exit(1)
-      end
+      return error(result) if result.error?
+
+      [command_builder(command), result.arguments]
+    end
 
-      [command.new, result.arguments]
+    # Prepares command instance and injects context
```

Comment:
"injects context" part is no longer relevant given you've reduced the scope of this PR?